// grunt-cli/Gruntfile.js
/*
 * grunt-cli
 * http://gruntjs.com/
 *
 * Copyright (c) 2012 Tyler Kellen, contributors
 * Licensed under the MIT license.
 * https://github.com/gruntjs/grunt-init/blob/master/LICENSE-MIT
 */

'use strict';

module.exports = function(grunt) {

  // Project configuration.
  grunt.initConfig({
    

    less: {
        compile: {
            files: {
                '../../css/style.css': '../../css/less/style.less'
            }
        },
        yuicompress: {
            files: {
                '../../css/style.min.css': '../../css/style.css'
            },
            options: {
                yuicompress: true
            }
        }
    },
    watch: {
        scripts: {
            files: ['css/*.less'],
            tasks: ['less']
        }
    },
  });

  //less 打包
  grunt.loadNpmTasks('grunt-contrib-less');

  //代码压缩
  grunt.loadNpmTasks('grunt-contrib-cssmin');
  //浏览器前缀补全
  grunt.loadNpmTasks('grunt-autoprefixer');
  //
  grunt.loadNpmTasks('grunt-contrib-less');

  // These plugins provide necessary tasks.
 // grunt.loadNpmTasks('grunt-contrib-jshint');

  // "npm test" runs these tasks
  grunt.registerTask('test', ['less']);

  // Default task.
  grunt.registerTask('default', ['less']);

};
